File: facade/myfacade.cpp
#include <iostream>

using namespace std;

class ColdFood {
    public:
    string WashRinse() const {
        return "Cold Food: Wash and rinse";
    }
    string Wrap() const {
        return "Wrap in a box or bag";
    }
    string Freeze() const {
        return "Freeze in a refrigerator";
    }
};

class HotFood {
    public:
    string Unwrap() const {
        return "Take the food out of bag";
    }
    string Clean() const {
        return "Clean the food";
    }
    string Cook() const {
        return "Cook in the heat";
    }
};

class Restaurant {
    protected:
    ColdFood *coldfood;
    HotFood *hotfood;

    public:
    Restaurant(ColdFood *coldfood = nullptr,
               HotFood *hotfood = nullptr) {
                this->coldfood = coldfood ?: new ColdFood;
                this->hotfood = hotfood ?: new HotFood;
               }
    ~Restaurant() {
        delete coldfood;
        delete hotfood;
    }
    string Operation() {
        string result = "Restaurant Opens..\n";
        result += coldfood->WashRinse() + "\n";
        result += hotfood->Unwrap() + "\n";

        return result;
    }
};

void client(Restaurant *restaurant) {
    cout << restaurant->Operation();
}

int main() {
    ColdFood *coldfood = new ColdFood;
    HotFood *hotfood = new HotFood;

    Restaurant *restaurant = new Restaurant(coldfood, hotfood);
    client(restaurant);

    return 0;
}

File: facade/swift/myfacade.swift
class ColdFood {
    func WashAndRinse() -> String {
        return("ColdFood : Wash and Rinse")
    }

    func Wrap() -> String {
        return("ColdFood ; Wrap")
    }

    func Freeze() -> String {
        return("ColdFood : Freeze")
    }
}

class HotFood {
    func UnWrap() -> String {
        return("HotFood : Unwrap")
    }

    func Clean() -> String {
        return("HotFood : Clean")
    }

    func Cook() -> String {
        return("HotFood : Cook")
    }
}

class Restaurant {
    var coldFood = ColdFood()
    var hotFood = HotFood()

    init(coldFood: ColdFood, hotFood: HotFood) {
        self.coldFood = coldFood
        self.hotFood = hotFood
    }

    func Operation() -> String {
        var result = String()

        result += coldFood.WashAndRinse()
        result += coldFood.Wrap()
        result += coldFood.Freeze()

        result += hotFood.UnWrap()
        result += hotFood.Clean()
        result += hotFood.Cook()

        return(result)
    }
}

@main
class MyFacade {
    static func main() {
        let coldFood = ColdFood()
        let hotFood = HotFood()

        let restaurant = Restaurant(coldFood: coldFood, hotFood: hotFood)
        print(restaurant.Operation())
    }
}